sdk/include/opentelemetry/sdk/common/attribute_utils.h

```diff
@@ -162,8 +216,36 @@ class AttributeMap : public std::unordered_map<std::string, OwnedAttributeValue>
     (*this)[std::string(key)] = nostd::visit(converter_, value);
   }
 
+  bool EqualTo(const opentelemetry::common::KeyValueIterable &attributes) const noexcept
+  {
+    if (attributes.size() != this->size())
+    {
+      return false;
+    }
+
+    const bool is_equal = attributes.ForEachKeyValue(
+        [this](nostd::string_view key,
+               const opentelemetry::common::AttributeValue &value) noexcept {
+          // Perform a linear search to find the key assuming the map is small
+          // This avoids temporary string creation from this->find(std::string(key))
+          for (const auto &kv : *this)
+          {
+            if (kv.first == key)
+            {
+              // Order of arguments is important here. OwnedAttributeValue is first then
+              // AttributeValue AttributeEqualToVisitor does not support the reverse order
+              return nostd::visit(equal_to_visitor_, kv.second, value);
+            }
+          }
+          return false;
+        });
+
+    return is_equal;
+  }
+
 private:
   AttributeConverter converter_;
+  AttributeEqualToVisitor equal_to_visitor_;
```

Comment:
Could the visitor be a local object in the `EqualTo()` method, instead of a member of the class ?

This would reduce the memory footprint of the attribute map.

Reply:
Right it is a struct with no data. Follows the existing style as the converter.